dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/entity/DataSource.java

```diff
@@ -75,6 +71,17 @@ public class DataSource {
      */
     private Date updateTime;
 
+    /**
+     * test flag
+     */
+    protected int testFlag;
+
+    /**
+     * bind test data source id
+     */
+    @TableField(fill = FieldFill.INSERT_UPDATE)
```

Comment:
When converting the online data source to the test data source, the bind passed to the back end_ test_ ID is null. Since mybatis plus is used for updating, you will find that the update statement will omit the field with null value, which will cause problems. Therefore, the function of this annotation is that even if the value is null, it cannot be omitted.